### PR TITLE
Add rule for super method with parens

### LINKED ISSUE #703: The super method, parentheses with arguments?
For the `super` method being a keyword in ruby, I am curious as to if it requires parentheses around arguments or not. I am having trouble grasping if it is declarative or not.

From my understanding these are correct:

```
def something_1
  super
end

def something_2
  super(:foo, :bar, :baz)
end

def something_3
  super :foo, :bar, :baz
end
```
 In terms of `something_2` vs `something_3` is one better over the other?

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #703.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -2442,6 +2442,22 @@ puts temperance.age
 system 'ls'
 ----
 
+==== "super" method usage [[super-parens]]
+
+Always use parenthesis when calling `super` with arguments (keyword or not):
```
Comment (bbatsov) [inline README.adoc]: parentheses

File: README.adoc [hunk 2]
```diff
@@ -2442,6 +2442,22 @@ puts temperance.age
...
+
+[source,ruby]
+----
+# bad
+super name, age
+super :name, :age
```
Comment (bbatsov) [inline README.adoc]: Why do symbols warrant a special case?

File: README.adoc [hunk 3]
```diff
@@ -2442,6 +2442,22 @@ puts temperance.age
...
+# good
+super(name, age)
+super(:name, :age)
+----
+
+When calling `super` without arguments, `super` and `super()` mean different things. Decide what is appropriate for your usage.
```
Comment (bbatsov) [inline README.adoc]: I'd put `IMPORTANT:` in front of this (an admonition).

**Comment by tejasbubane** [comment]
@bbatsov Fixed.

**Pull request merged**